### PR TITLE
[CI] Improve validation python script

### PR DESCRIPTION
**Pull request by binarymaster**
- Add checks for section duplicates (happened in bcadc024)
- Add checks for required section keys (happened in 7001cf34)
- Add checks for value duplicates (happened in 7001cf34)
- Also fix URL to the RAPPS wiki

### CODE REVIEW

**Comment by binarymaster** [comment]
Yes, CI is expected to return failure now, because master is broken.
```
maelstorm.txt(1:0): Main section has no b'Category' key!
b'[Section]\r\n'
 ^
sopwith3.txt(2:0): Duplicate value found: b'Maelstorm, version 3.0.6'
b'Name = Maelstorm\r\n'
 ^
maelstorm.txt(2:0): First occurence:
b'Name = Maelstorm\r\n'
 ^
Please check https://reactos.org/wiki/RAPPS for details on the file format.
```

File: .github/validate-rapps.py [hunk 1]
```diff
@@ -39,6 +46,12 @@
 ]
 
 
+all_name_lines = []
+all_url_lines = []
+all_names = []
+all_urls = []
```
Comment (learn-more) [inline .github/validate-rapps.py:52]: If you make these things a dictionary, you can lookup / use them easier.
f.e.:

```
# Define:
all_names = {}
# Lookup (note the 'None' default argument, without that this throws)
first = all_names.get(name, None)
# Use it
if first is not None:
    print(f'Duplicate shit, first occurrence: {first}')
else:
    all_names[name] = line
```

File: .github/validate-rapps.py [hunk 2]
```diff
@@ -167,6 +185,14 @@ def __init__(self, fullname):
         self.filename = os.path.basename(fullname)
         self._sections = []
 
+    def verify_unique(self, reporter, lines, line, names, name):
```
Comment (learn-more) [inline .github/validate-rapps.py:188]: This does not need to be a class function, because all arguments are provided.

**Review by learn-more — COMMENTED** [review]
Some minor comments

File: .github/validate-rapps.py (hunk 1 above)
Comment (binarymaster) [reply .github/validate-rapps.py]: Addressed, thanks! :)

**Pull request merged**